app/src/parser_impl.c

```diff
@@ -38,6 +38,18 @@ parser_error_t _read(parser_context_t *ctx, parser_tx_t *v) {
     return parser_ok;
 }
 
+bool hasMemoToPrint(const parser_context_t *ctx) {
+    // Check if memoSection exists and has a commitmentDiscriminant
+    if ((ctx->tx_obj->transaction.header.memoSection != NULL && 
```

Comment:
Maybe no need for the if? and just return the (comparisons),however this way is a bit more explicit